zcash_client_sqlite/src/wallet/init/migrations/add_transaction_views.rs

```diff
@@ -0,0 +1,356 @@
+                   LEFT JOIN blocks
+                          ON transactions.block = blocks.height
+            GROUP BY received_notes.tx;
+            CREATE VIEW v_transactions AS
+            SELECT id_tx,
+                   mined_height,
+                   tx_index,
+                   txid,
+                   expiry_height,
+                   raw,
+                   SUM(value) + MAX(fee) AS net_value,
+                   SUM(is_change) > 0 AS has_change,
+                   SUM(memo_present) AS memo_count
+            FROM (
+                SELECT transactions.id_tx            AS id_tx,
+                       transactions.block            AS mined_height,
+                       transactions.tx_index         AS tx_index,
+                       transactions.txid             AS txid,
+                       transactions.expiry_height    AS expiry_height,
+                       transactions.raw              AS raw,
+                       0                             AS fee,
+                       CASE
+                            WHEN received_notes.is_change THEN 0
+                            ELSE value
+                       END AS value,
+                       received_notes.is_change      AS is_change,
+                       CASE
+                           WHEN received_notes.memo IS NULL THEN 0
+                           WHEN SUBSTR(received_notes.memo, 0, 2) = X'F6' THEN 0
+                           ELSE 1
+                       END AS memo_present
+                FROM   transactions
+                       JOIN received_notes ON transactions.id_tx = received_notes.tx
+                UNION
+                SELECT transactions.id_tx            AS id_tx,
+                       transactions.block            AS mined_height,
+                       transactions.tx_index         AS tx_index,
+                       transactions.txid             AS txid,
+                       transactions.expiry_height    AS expiry_height,
+                       transactions.raw              AS raw,
+                       transactions.fee              AS fee,
+                       -sent_notes.value             AS value,
+                       false                         AS is_change,
+                       CASE
+                           WHEN sent_notes.memo IS NULL THEN 0
+                           WHEN SUBSTR(sent_notes.memo, 0, 2) = X'F6' THEN 0
+                           ELSE 1
+                       END AS memo_present
+                FROM   transactions
+                       JOIN sent_notes ON transactions.id_tx = sent_notes.tx
+            )
+            GROUP BY id_tx;",
+        )?;
+
+        Ok(())
+    }
+
+    fn down(&self, transaction: &rusqlite::Transaction) -> Result<(), WalletMigrationError> {
+        transaction.execute_batch(
+            "DROP VIEW v_tx_sent_notes;
+            DROP VIEW v_tx_received_notes;
+            DROP VIEW v_tx_notes;",
+        )?;
+        Ok(())
```

Comment:
This needs to also remove the `fee` column in order to be a correct down-migration. That is quite involved however. There are several possible alternatives:
- Remove the column via "create table without `fee` column, copy all data across, delete old table, rename new table".
- Alter the up-migration to correctly handle the case where the `fee` column exists:
  - If the column doesn't exist, add it and fill it.
  - If the column does exist, check its contents matches what we would have filled in.
- Make the entire down-migration panic (the "simple" solution).